### PR TITLE
Die more neatly if synapse fails to start

### PR DESCRIPTION
**Pull request by richvdh**
Try to avoid waiting for something that will never happen if synapse never
starts.

### CODE REVIEW

File: tests/05synapse.pl [hunk 1]
```diff
@@ -28,11 +28,15 @@ sub extract_extra_args
 # see
 #   https://rt.perl.org/Public/Bug/Display.html?id=128774
 main::AT_END sub {
-   $OUTPUT->diag( "Killing synapse servers " ) if @synapses;
 
    ( fmap_void {
       my $synapse = $_;
 
+      # skip this if the process never got started.
+      return Future->done unless $synapse->{proc};
```
Comment (leonerd) [inline tests/05synapse.pl]: That's a little bit of gut-wrenching into the innards of the `$synapse` object there. I'd use the `->pid` accessor method that you've already fixed to return zero if there isn't a process:

    return Future->done unless $synapse->pid;

Comment (richvdh) [inline tests/05synapse.pl]: seems fair. done in dee61df

File: tests/05synapse.pl [hunk 2]
```diff
@@ -28,11 +28,15 @@ sub extract_extra_args
 # see
 #   https://rt.perl.org/Public/Bug/Display.html?id=128774
 main::AT_END sub {
-   $OUTPUT->diag( "Killing synapse servers " ) if @synapses;
 
    ( fmap_void {
       my $synapse = $_;
 
+      # skip this if the process never got started.
+      return Future->done unless $synapse->{pid};
```
Comment (leonerd) [inline tests/05synapse.pl]: Surely `$synapse->pid`. It's a method not an internal hash slot access.

Comment (richvdh) [inline tests/05synapse.pl]: oh bugger

**Comment by leonerd** [comment]
Much better. LGTM

**Pull request merged**